quickannotator/api/v1/tile/routes.py

```diff
@@ -0,0 +1,124 @@
+from quickannotator.db import db_session
+from quickannotator.db.crud.tile import TileStoreFactory, TileStore
+from quickannotator.constants import TileStatus
+import quickannotator.db.models as db_models
+from . import models as server_models
+from quickannotator.api.v1.utils.coordinate_space import get_tilespace
+from flask_smorest import Blueprint
+
+bp = Blueprint('tile', __name__, description="Tile operations")
+
+@bp.route('/<int:image_id>/<int:annotation_class_id>')
+class Tile(MethodView):
+    @bp.arguments(server_models.GetTileArgsSchema, location='query')
+    @bp.response(200, server_models.TileRespSchema)
+    def get(self, args, image_id, annotation_class_id):
+        """     returns a Tile
+        """
+        tilestore = TileStoreFactory.get_tilestore()
+        result = tilestore.get_tile(image_id, annotation_class_id, args['tile_id'])
+        return result, 200
+
+    @bp.arguments(server_models.GetTileArgsSchema, location='query')
+    def delete(self, args, image_id, annotation_class_id):
+        """     delete a Tile
+        """
+        db_session.query(db_models.Tile).filter_by(
+            image_id=image_id,
+            annotation_class_id=annotation_class_id,
+            tile_id=args['tile_id']
+        ).delete()
+        return 204
+
+@bp.route('/<int:image_id>/<int:annotation_class_id>/predict')
+class PredictTile(MethodView):
+    @bp.arguments(server_models.PostTileArgsSchema, location='query')
+    @bp.response(200, server_models.TileRespSchema, description="Staged tile for DL processing")
+    def post(self, args, image_id, annotation_class_id):
+        """     stage a tile for DL processing     """
+        tilestore: TileStore = TileStoreFactory.get_tilestore()
+        result = tilestore.upsert_pred_tiles(
+            image_id=image_id,
+            annotation_class_id=annotation_class_id,
+            tile_ids=[args['tile_id']],
+            pred_status=TileStatus.STARTPROCESSING,
+            process_owns_tile=False
+        )
+
+        if len(result) > 0:
+            return result[0], 200
+        else:
+            return {"message": "Failed to stage tile for processing"}, 400
+
+@bp.route('/<int:image_id>/<int:annotation_class_id>/bbox')
+class TileBoundingBox(MethodView):
+    @bp.arguments(server_models.GetTileArgsSchema, location='query')
+    @bp.response(200, server_models.TileBoundingBoxRespSchema)
+    def get(self, args, image_id, annotation_class_id):
+        """     get the bounding box for a given tile as a GeoJSON polygon
+        """
+        tilespace = get_tilespace(image_id=image_id, annotation_class_id=annotation_class_id, in_work_mag=False)
+        bbox = tilespace.get_bbox_for_tile(args['tile_id'])
+        geojson_polygon = {
+            "type": "Polygon",
+            "coordinates": [[
+                [bbox[0], bbox[1]],
+                [bbox[2], bbox[1]],
+                [bbox[2], bbox[3]],
+                [bbox[0], bbox[3]],
+                [bbox[0], bbox[1]]
```

Comment:
interetsing - does the polygon need to be explicitly closed? i would not have thought to do that

Reply:
Yes, all geojson polygons must be closed.